### PR TITLE
Allow a popup page as the download page

### PR DESCRIPTION
**Pull request by EBusch**

### CODE REVIEW

File: Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs [hunk 1]
```diff
@@ -9,6 +9,9 @@
 using Float.TinCan.LocalLRSServer;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Rg.Plugins.Popup.Extensions;
+using Rg.Plugins.Popup.Pages;
+using Rg.Plugins.Popup.Services;
```
Comment (mediabounds) [inline Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs]: @EBusch Ideally we would avoid adding new dependencies to this library. We can do that by creating new overridable methods for handling presenting and dismissing of the downloading page.

File: Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs [hunk 2]
```diff
@@ -128,7 +132,16 @@ public override void Start()
...
+                    var statusPage = CreateDownloadStatusPage(downloadStatus);
+                    if (statusPage is PopupPage popupPage)
+                    {
+                        downloadPopupPage = popupPage;
+                        PopupNavigation.Instance.PushAsync(downloadPopupPage, true);
+                    }
+                    else
+                    {
+                        NavigationContext.PresentPage(statusPage);
+                    }
```
Comment (mediabounds) [inline Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs]: @EBusch ActivityLibrary doesn't need to care about the type of page -- we can instead do something like this:
```
ShowDownloadStatus(CreateDownloadStatusPage(downloadStatus));
```

And define this method:
```
protected virtual void ShowDownloadStatus(ContentPage downloadStatusPage)
{
  NavigationContext.PresentPage(downloadStatusPage);
}
```

Now the child class can override that method and decide how to show the page.

File: Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs [hunk 3]
```diff
@@ -342,7 +355,15 @@ protected virtual async void HandleDownloadCompleted(object sender, EventArgs ar
...
-                await NavigationContext.DismissPageAsync();
+                if (downloadPopupPage != null)
+                {
+                    await PopupNavigation.Instance.RemovePageAsync(downloadPopupPage, true);
+                    downloadPopupPage = null;
+                }
+                else
+                {
+                    await NavigationContext.PopPageAsync();
+                }
```
Comment (mediabounds) [inline Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs]: @EBusch Similarly, we can do something like this here:
```
await HideDownloadStatus();
...
protected virtual Task HideDownloadStatus()
{
  return NavigationContext.PopPageAsync();
}
```

**Review by mediabounds — CHANGES_REQUESTED** [review]

File: Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs [hunk 4]
```diff
@@ -127,8 +128,7 @@ public override void Start()
 
                     downloadStatus.DownloadsCompleted += HandleDownloadCompleted;
                     downloadStatus.DownloadsCancelled += HandleDownloadCancelled;
-
-                    NavigationContext.PresentPage(CreateDownloadStatusPage(downloadStatus));
+                    await ShowDownloadStatus(CreateDownloadStatusPage(downloadStatus));
```
Comment (bbeversdorf) [inline Float.TinCan.ActivityLibrary/ActivityLaunchCoordinator.cs:131]: @EBusch Could we get into a state where we are creating/presenting the download page more than once?
_just wondering if we need a protection from creating multiple_

**Review by bbeversdorf — APPROVED** [review]
In general, this looks(?) pretty safe. It will need a thorough run thru though.

**Pull request merged**